data/delegation/store.go

```diff
@@ -0,0 +1,391 @@
+package delegation
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/Oneledger/protocol/data/balance"
+	"github.com/Oneledger/protocol/data/keys"
+	"github.com/Oneledger/protocol/serialize"
+	"github.com/Oneledger/protocol/storage"
+)
+
+type DelegationStore struct {
+	state  *storage.State
+	szlr   serialize.Serializer
+	prefix []byte
+	mux    sync.Mutex
+}
+
+type MatureBlock struct {
+	Height int64
+	Data   []*MatureData
+}
+
+type MatureData struct {
+	Address keys.Address
+	Amount  balance.Amount
+	Height  int64
+}
+
+func NewDelegationStore(prefix string, state *storage.State) *DelegationStore {
+	return &DelegationStore{
+		state:  state,
+		prefix: storage.Prefix(prefix),
+		szlr:   serialize.GetSerializer(serialize.PERSISTENT),
+	}
+}
+
+func (st *DelegationStore) WithState(state *storage.State) *DelegationStore {
+	st.state = state
+	return st
+}
+
+func (st *DelegationStore) Get(key []byte) (amt *balance.Amount, err error) {
+	prefixKey := append(st.prefix, key...)
+
+	dat, _ := st.state.Get(storage.StoreKey(prefixKey))
+	amt = balance.NewAmount(0)
+	if len(dat) == 0 {
+		return
+	}
+	err = serialize.GetSerializer(serialize.PERSISTENT).Deserialize(dat, amt)
+	return
+}
+
+func (st *DelegationStore) Set(key []byte, amt *balance.Amount) error {
+	dat, err := serialize.GetSerializer(serialize.PERSISTENT).Serialize(amt)
+	if err != nil {
+		return err
+	}
+
+	prefixKey := append(st.prefix, key...)
+	err = st.state.Set(storage.StoreKey(prefixKey), dat)
+	return err
+}
+
+// Validator - Delegator data
+
+func (st *DelegationStore) getVDKey(validatorAddress keys.Address, delegatorAddress keys.Address) []byte {
+	key := []byte(fmt.Sprintf("_e_%s_%s", validatorAddress, delegatorAddress))
```

Comment:
Can we use String() method to convert address to string? Because printing addresses as %s is not human readable, and this conversion helps trouble shooting.

Reply:
`fmt.Println(fmt.Sprintf("_e_%s_%s", validatorAddress, delegatorAddress))`
`_e_0lt908566b0c536e011ae4f4c25f67bf3c55ef38a26_0lt9d35e060a5fddcc200551d5b4a70efabb31483bb`

Reply:
seems like it human readable as Address has implementation of String()